File: bigbluebutton-apps/src/main/scala/org/bigbluebutton/core/apps/poll/Response.scala
package org.bigbluebutton.core.apps.poll

import scala.collection.mutable.ArrayBuffer

case class Responder(val userID: String, val name: String)

class Response(val id: Int, val response: String) {

  val responders = new ArrayBuffer[Responder]()
  
  def addResponder(responder: Responder) {
	responders += responder
  }
  
  def numResponders():Int = {
    responders.length;
  }
  
  def getResponders():Array[Responder] = {
    var r = new Array[Responder](responders.length)
    responders.copyToArray(r)
    return r
  }
}

File: bigbluebutton-apps/src/main/java/org/bigbluebutton/core/apps/poll/model/Response.java
package org.bigbluebutton.core.apps.poll.model;

import java.util.ArrayList;

public class Response {
	private final String id;
	private final String response;
	private final ArrayList<Responder> responders = new ArrayList<Responder>();
	
	public Response(String id, String response) {
		this.id = id;
		this.response = response;
	}
	
	public String getID() {
		return id;	
	}
	
	public String getResponse() {
		return response;
	}
	
	public void addResponder(Responder resp) {
		responders.add(resp);
	}
	
	public ArrayList<String> getResponders() {
		ArrayList<String> resps = new ArrayList<String>();
		
		for (Responder r : responders) {
			resps.add(r.getUserID());
		}
		
		return resps;
	}
	
	public Response copy() {
		Response resp = new Response(id, response);
		ArrayList<String> resps = getResponders();
		
		for (String userID : resps) {
			resp.addResponder(new Responder(userID));
		}
		
		return resp;
	}
}
